### PR TITLE
EE-1245: revert EE-1205

### PR DESCRIPTION
**Pull request by henrytill**
Ref: 
https://github.com/CasperLabs/casper-node/pull/1262

### CODE REVIEW

**Review by dwerner — APPROVED** [review]

File: types/src/execution_result.rs [hunk 1]
```diff
@@ -571,16 +562,7 @@ impl ToBytes for Transform {
...
-            Transform::WriteWithdraw(value) => value.serialized_length() + U8_SERIALIZED_LENGTH,
-            Transform::WriteEraValidators(value) => {
-                value.serialized_length() + U8_SERIALIZED_LENGTH
-            }
-            Transform::Identity
-            | Transform::WriteContractWasm
-            | Transform::WriteContract
-            | Transform::WriteContractPackage
-            | Transform::Delete => U8_SERIALIZED_LENGTH,
+            _ => U8_SERIALIZED_LENGTH,
```
Comment (Fraser999) [inline types/src/execution_result.rs:565]: This change looks wrong.  We should also probably avoid using a catchall case here and explicitly list every variant.

File: types/src/execution_result.rs [hunk 2]
```diff
@@ -652,7 +633,7 @@ impl FromBytes for Transform {
 impl Distribution<Transform> for Standard {
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Transform {
         // TODO - include WriteDeployInfo and WriteTransfer as options
-        match rng.gen_range(0..14) {
+        match rng.gen_range(0..13) {
```
Comment (Fraser999) [inline types/src/execution_result.rs:636]: I guess not for this PR, but it's worrying that we're not covering all variants here.  We use this in the bytesrepr roundtrip test, so only 13 of the 19 variants are being tested.

**Review by Fraser999 — CHANGES_REQUESTED** [review]
Just a simple bug in the `Transform` code blocking this.

File: types/src/execution_result.rs (hunk 1 above)
Comment (dwerner) [reply types/src/execution_result.rs:565]: This is supposed to be a revert PR only, so we should put that in a separate PR.

Comment (henrytill) [reply types/src/execution_result.rs:565]: Agreed, I would like to do this in a separate PR.  These are purely reverting things to where they were.

**Review by mpapierski — APPROVED** [review]

**Review by EdHastingsCasperLabs — APPROVED** [review]

File: types/src/execution_result.rs (hunk 1 above)
Comment (EdHastingsCasperLabs) [reply types/src/execution_result.rs:565]: double plus agreed

Comment (Fraser999) [reply types/src/execution_result.rs:565]: Sure - I'll approve on that assumption

**Review by Fraser999 — APPROVED** [review]

**Pull request merged**